muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java

```diff
@@ -1075,6 +1076,16 @@ public Response listWorkspaceStaffMembers(@PathParam("ID") Long workspaceEntityI
       
       if (user != null) {
         UserEntity userEntity = userEntityController.findUserEntityByUser(user);
+        
+        // #3111: Workspace staff members should be limited to teachers only. A better implementation would support specified workspace roles
+        
+        if (workspaceEntity != null) {
```

Comment:
Yup, the null check is unnecessary. Other than that, this has the same logic as above. This method is used in the view that shows all the teachers and students of a workspace.

Later on, we should probably refactor the code so that all views would use the same method to retrieve the staff members of a workspace.